### PR TITLE
Add XLSX Excel As Export Option

### PR DESCRIPTION
**Pull request by mwildgoose**
The current Excel export option doesn't actually generate an Excel file, it generates a HTML file with the XLS extension.
This produces a warning as noted in issue #64.

This pull request adds a new Excel XLSX export option and rebrands the old XLS option as a HTML export.

This allows for using Excel-specific features, such as:
- Using a freeze-pane on the headers
- Grouping rows to make collapsible sections

new_export_excel

I've aligned the export buttons into a single row so it doesn't take up too much space in the UI (as the column list is already overflowing off the page):
| Old Export Box | New Export Box |
| --- | --- |
| old_export_box | new_export_box |

### CODE REVIEW

**Comment by crobibero** [comment]
I appreciate the effort that went into this PR. I would prefer to use a library to write actual excel files because the base xml seems like it will be very difficult to maintain.

Preferred library: https://github.com/ClosedXML/ClosedXML
If that doesn't work: https://github.com/EPPlusSoftware/EPPlus (not completely FOSS license)

File: build.yaml [hunk 1]
```diff
@@ -11,6 +11,15 @@ description: "Generate reports of your media library"
...
   - "Jellyfin.Plugin.Reports.dll"
+  - "ClosedXML.dll"
+  - "DocumentFormat.OpenXml.dll"
+  - "ExcelNumberFormat.dll"
+  - "Microsoft.Win32.SystemEvents.dll"
+  - "System.Drawing.Common.dll"
+  - "System.IO.Packaging.dll"
+  - "runtimes/unix/lib/netcoreapp2.0/System.Drawing.Common.dll"
+  - "runtimes/win/lib/netcoreapp2.0/Microsoft.Win32.SystemEvents.dll"
+  - "runtimes/win/lib/netcoreapp2.0/System.Drawing.Common.dll"
```
Comment (crobibero) [inline build.yaml:22]:
```suggestion
  - "System.IO.Packaging.dll"
```

The removed are already bundled with Jellyfin. Unsure if System.IO.Packaging is required- will hopefully have time to test later

File: Jellyfin.Plugin.Reports/Api/Data/ReportExport.cs [hunk 2]
```diff
@@ -12,231 +15,192 @@ public static class ReportExport
...
+                worksheet.Cell(nextRow, 1).InsertData(rows);
+                nextRow += rows.Count();
+            }
 
-            Html += returnValue.ToString();
-            Html += "</body>";
-            Html += "</html>";
-            return Html;
-        }
+            IXLWorkbook workbook = new XLWorkbook(XLEventTracking.Disabled);
```
Comment (crobibero) [inline Jellyfin.Plugin.Reports/Api/Data/ReportExport.cs:156]:
```suggestion
            using IXLWorkbook workbook = new XLWorkbook(XLEventTracking.Disabled);
```

File: Jellyfin.Plugin.Reports/Api/Data/ReportExport.cs [hunk 3]
```diff
@@ -12,231 +15,192 @@ public static class ReportExport
...
+            // Workbook properties
+            workbook.Properties.Author = "Jellyfin";
+            workbook.Properties.Title = "ReportExport";
+            string pluginVer = Assembly.GetExecutingAssembly().GetName().Version.ToString();
+            workbook.Properties.Comments = $"Produced by Jellyfin Reports Plugin {pluginVer}";
+
+            // Save workbook to stream and return
+            MemoryStream memoryStream = new MemoryStream();
+            workbook.SaveAs(memoryStream);
+            workbook.Dispose();
```
Comment (crobibero) [inline Jellyfin.Plugin.Reports/Api/Data/ReportExport.cs:201]:
```suggestion
```

**Comment by mwildgoose** [comment]
Thanks for the feedback, ClosedXML does make the code a lot cleaner.
Only downsides are that it's increased the plugin size from 170KB to 8.7MB and the ClosedXML column autofit implementation doesn't work on Linux (see [this issue](https://github.com/ClosedXML/ClosedXML/issues/1283)) so I've commented that out for now.

**Comment by crobibero** [comment]
Bummer that the feature doesn't work.. but I confirmed that the plugin works as expected. Thanks!

**Review by crobibero — APPROVED** [review]

**Pull request merged**